compiler/rustc_middle/src/middle/region.rs

```diff
@@ -362,6 +362,10 @@ impl ScopeTree {
         self.parent_map.get(&id).cloned().map(|(p, _)| p)
     }
 
+    pub fn var_has_scope(&self, var_id: hir::ItemLocalId) -> bool {
```

Comment:
Comment would be good!

Reply:
I entirely removed the function, for a better(?) approach and added a comment explaining why we are doing a check.